x/auth/ante.go

```diff
@@ -95,8 +95,15 @@ func NewAnteHandler(am AccountMapper, fck FeeCollectionKeeper) sdk.AnteHandler {
 			}
 
 			// first sig pays the fees
-			// TODO: Add min fees
 			// Can this function be moved outside of the loop?
+			if ctx.IsCheckTx() && !simulate {
+				// Required fees must be greater than the minimum set by the validator
+				if ctx.MinimumFees().Minus(fee.Amount).IsNotNegative() {
+					fee = NewStdFee(fee.Gas, ctx.MinimumFees()...)
+				}
+				// Validators reject any tx from the mempool with less than the required fee per gas
```

Comment:
`validators`

Reply:
Where does the rejection actually occur? If the fee amount is zero?

Reply:
> validators

ACKd.

> Where does the rejection actually occur? If the fee amount is zero?

Currently the rejection occurs at `deductFees()`.